File: ios/Classes/PassageFlutterExtensions.swift
import Passage

internal func dictToJsonString(_ dict: [String: Any]) -> String {
    let jsonData = try! JSONSerialization.data(withJSONObject: dict, options: [])
    let jsonString = String(data: jsonData, encoding: .utf8)!
    return jsonString
}

internal extension AuthResult {
    
     func toDictionary() -> [String: Any] {
         var authResultDict: [String : Any] = [
             "authToken": authToken,
             "redirectUrl": redirectURL,
             "refreshToken": refreshToken,
             "refreshTokenExpiration": refreshTokenExpiration
         ]
         return authResultDict
     }
    
    func toJsonString() -> String {
        return dictToJsonString(toDictionary())
    }
    
 }

File: android/src/main/kotlin/id/passage/passage_flutter/PassageFlutterExtensions.kt
package id.passage.passage_flutter

import id.passage.android.model.AuthResult

internal fun AuthResult.convertToMap(): HashMap<String, Any> {
    val authResultHashMap:  HashMap<String, Any> = hashMapOf(
        "auth_token" to authToken,
        "redirect_url" to redirectUrl
    )
    refreshToken?.let { authResultHashMap["refresh_token"] = it }
    refreshTokenExpiration?.let { authResultHashMap["refresh_token_expiration"] = it }
    return authResultHashMap
}